### ISSUE TITLE
Allow to use any object as a choice

### ISSUE THREAD

**Issue by Bastian** [xdrop/fuzzywuzzy#44]
First of all, thank you for this great library.
However, there's a small issue I have with it: For one of my projects I'm implementing a search for JavaDoc methods and have a class `JavadocMethod` with methods like `getMethodName()`, `getClassName()` and `getUrl()`.
For searching it would be very convenient to just use the object itself for search, so I can access the url of the found method. 
I'm thinking about a generic solution like this:
```java
public static <T> List<ExtractedResult<T>> extractTop(String query, Collection<T> choices, Function<T, String> mapper, int limit)
```
which allows to use any object by just providing a function which maps this object to a string.
```java
Collection<JavadocMethod> methods = ...;
FuzzySearch.extractTop("String#valeuOf(loong)", methods, method -> String.format("%s#%s", method.getClassName(), method.getMethodName()), 5);
```

Can you imagine implementing such a feature or accept a pull requests that adds it?

**Comment by xdrop** [comment]
Hey, thanks I like your suggestion!

Ideally we implement this through additional overloads so as to not break the existing API. I'd be very grateful for a PR otherwise, I'll try and get the time to work on it (which may be a while)

**Comment by Bastian** [comment]
I'm glad you like it.

I'll try to create a PR when I find time for it.

**Comment by Bastian** [comment]
Sorry, but I don't know if I have time for this anytime soon.

**Comment by xdrop** [comment]
Closed in #44.
